torch/utils/data/datapipes/iter/combining.py

```diff
@@ -55,16 +57,97 @@ def __iter__(self):
 
 @functional_datapipe('fork')
 class ForkIterDataPipe(IterDataPipe):
+    r""" :class:`ForkIterDataPipe`.
+
+        Iterable DataPipe to create multiple instances of the same Iterable DataPipe.
+        args:
+            datapipe: Iterable DataPipe being copied
+            num_instances: number of instances of the datapipe to create
+            buffer_size: this restricts how far ahead the leading child DataPipe
+             can read relative to the slowest child DataPipe
+    """
+    def __new__(cls, datapipe: IterDataPipe, num_instances: int, buffer_size: int = 1000):
+        container = _ForkIterDataPipe(datapipe, num_instances, buffer_size)
+        return [ChildDataPipe(container, i) for i in range(num_instances)]
```

Comment:
This is likely the reason why it is not pickable, consider wrapping it into tuple like `tuple([...bla bla..])`

Reply:
Just tried that and still get the same error. Any suggestion?

```
test/test_datapipe.py:1406 (TestGraph.test_traverse_forked)
Traceback (most recent call last):
  File "/Users/ktse/pytorch/test/test_datapipe.py", line 1414, in test_traverse_forked
    graph = torch.utils.data.graph.traverse(combined_dp)
  File "/Users/ktse/pytorch/torch/utils/data/graph.py", line 43, in traverse
    d[datapipe].update(traverse(item))
  File "/Users/ktse/pytorch/torch/utils/data/graph.py", line 43, in traverse
    d[datapipe].update(traverse(item))
  File "/Users/ktse/pytorch/torch/utils/data/graph.py", line 43, in traverse
    d[datapipe].update(traverse(item))
  File "/Users/ktse/pytorch/torch/utils/data/graph.py", line 40, in traverse
    items = list_connected_datapipes(datapipe)
  File "/Users/ktse/pytorch/torch/utils/data/graph.py", line 34, in list_connected_datapipes
    p.dump(scan_obj)
TypeError: cannot pickle 'generator' object
~``

Reply:
In this case advice is to inspect returned objects dicts and see if there are any generators properties

Reply:
We determined this is not the cause of the pickle issue, so I am leaving it untouched.